src/git.rs

```diff
@@ -22,15 +27,41 @@ pub fn repo_hash<P: AsRef<Path>>(path: P) -> Option<String> {
     }
 }
 
-fn dirty<P: AsRef<Path>>(path: P) -> bool {
-    let path = path.as_ref().to_str();
-    let mut args = match path {
-        Some(path) => vec!["-C", path],
-        None => vec![],
-    };
-    args.extend(&["ls-files", "-m"]);
-    match Command::new("git").args(&args).output() {
+fn dirty(path: impl AsRef<Path>) -> bool {
+    match git_command(path.as_ref(), &["ls-files", "-m"]) {
         Ok(modified_files) => !modified_files.stdout.is_empty(),
         Err(_) => false,
     }
 }
+
+fn gitmodules_config(subcommand: &[&str]) -> Option<String> {
+    let mut args = vec!["config", "-f", ".gitmodules", "-z"];
+    args.extend(subcommand);
+    let output = git_command(Path::new("."), &args).ok()?;
+
+    if !output.status.success() {
+        return None;
+    }
+
+    let mut result = String::from_utf8(output.stdout).ok()?;
+    assert_eq!(result.pop(), Some('\0'));
+    Some(result)
+}
+
+// This file is also compiled from build.rs where this function is unused
+#[allow(dead_code)]
+pub(crate) fn repo_remote_url(path: impl AsRef<Path>) -> Option<String> {
+    // Find the subsection that defines the module for the given path:
+    let key_for_path = gitmodules_config(&[
+        "--name-only",
+        "--get-regexp",
+        r"submodule\..+\.path",
```

Comment:
Considering to replace `.` with `[\w-]` since we probably don't use any weird characters... On the other hand anything besides newlines and null char is allowed (assuming it is escaped properly):

https://git-scm.com/docs/git-config#:~:text=Subsection%20names%20are%20case,you%20don%E2%80%99t%20need%20to.

Reply:
Seems fine to me as is. We'll notice when it fails :)